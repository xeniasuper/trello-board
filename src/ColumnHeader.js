import React from 'react';

class ColumnHeader extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text : "",
            isNameAdded : false,
        }
    };

    handleChange = (e) => {
        this.setState({
            text : e.currentTarget.textContent,
        })
    };

    handleAddColumnName = () => {
        this.props.handleAddColumnClick();
        this.setState({
            isNameAdded : true,
        })
    };

    render() {
        let none = this.state.isNameAdded ?  " none" : "";
        let type = this.state.isNameAdded ? " header-text" : "card column-name";
        return (
            <div className="column-header">
                <div className={type}
                     contentEditable="true"
                     spellCheck="false"
                     onInput={this.handleChange}
                >
                </div>
                <button className={"add-column" + none}
                        onClick={this.handleAddColumnName}
                >
                    Добавить колонку
                </button>
                <div className="delete-icon remove-column"
                     onClick={this.props.handleRemoveColumn}
                ></div>
            </div>
        )
    }
}

export {ColumnHeader}
