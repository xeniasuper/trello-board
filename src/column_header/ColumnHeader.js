import React from "react";
import "./column_header.css"

class ColumnHeader extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text : "",
            isNameAdded : false,
        }
    };

    handleChange = (e) => {
        this.setState({
            text : e.currentTarget.textContent,
        })
    };

    handleAddColumnName = () => {
        this.props.handleAddColumnClick();
        this.setState({
            isNameAdded : true,
        })
    };

    componentDidMount(){
        this.textInput.focus();
    }

    render() {
        let none = this.state.isNameAdded ?  " none" : "";
        let type = this.state.isNameAdded ? " header-text" : "card column-name";
        return (
            <div className="column-header">
                <div className={type}
                     ref={(input) => { this.textInput = input; }}
                     contentEditable="true"
                     spellCheck="false"
                     placeholder="Введите название колонки"
                     onInput={this.handleChange}
                >
                </div>
                <button className={"add-column" + none}
                        onClick={this.handleAddColumnName}
                >
                    Добавить колонку
                </button>
                <div className="delete-icon remove-column"
                     onClick={this.props.handleRemoveColumn}
                ></div>
            </div>
        )
    }
}

export {ColumnHeader}
