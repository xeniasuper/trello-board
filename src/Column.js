import React from "react";
import {Cards} from "./Cards"
import {ColumnHeader} from "./ColumnHeader";

class Column extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id : this.props.id,
            isNameAdded : false,
            cards : null,
        }
    }

    handleAddColumnClick = () => {
        this.setState({
            isNameAdded : true,
            cards : <Cards />
        })
    };

    render() {
        return (
            <div className="column">
                <ColumnHeader handleAddColumnClick={this.handleAddColumnClick}
                              isAdded={this.state.isNameAdded}
                              handleRemoveColumn={() => this.props.removeColumn(this.state.id)}
                />
                {/*<Cards visible={this.state.isNameAdded}/>*/}
                {this.state.cards}
            </div>
        )
    }
}

export {Column}
