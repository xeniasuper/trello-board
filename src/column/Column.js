import React from "react";
import {Cards} from "../cards/Cards"
import {ColumnHeader} from "../column_header/ColumnHeader";
import "./column.css";

class Column extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id : this.props.id,
            isNameAdded : false,
            cards : null,
        }
    }

    handleAddColumnClick = () => {
        this.setState({
            isNameAdded : true,
            cards : <Cards />
        })
    };

    render() {
        return (
            <div className="column">
                <ColumnHeader handleAddColumnClick={this.handleAddColumnClick}
                              isAdded={this.state.isNameAdded}
                              handleRemoveColumn={() => this.props.removeColumn(this.state.id)}
                />
                {this.state.cards}
            </div>
        )
    }
}

export {Column}
