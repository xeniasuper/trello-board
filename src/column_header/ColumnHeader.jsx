import React from 'react';
import PropTypes from 'prop-types';
import './column_header.css';

export default class ColumnHeader extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
      isNameAdded: false,
    };
  }

  handleChange = (e) => {
    this.setState({
      text: e.currentTarget.textContent,
    });
  };

  handleAddColumnName = () => {
    const { handleAddColumnClick } = this.props;
    handleAddColumnClick();
    this.setState({
      isNameAdded: true,
    });
  };

  componentDidMount = () => {
    this.textInput.focus();
  };

  render() {
    const { handleRemoveColumn } = this.props;
    const { isNameAdded } = this.state;
    const none = isNameAdded ? 'none' : '';
    const type = isNameAdded ? 'header-text' : 'card column-name';

    return (
      <div className="column-header">
        <div
          className={type}
          ref={(input) => { this.textInput = input; }}
          contentEditable="true"
          spellCheck="false"
          placeholder="Введите название колонки"
          onInput={this.handleChange}
        />
        <button
          type="button"
          className={'add-column ' + none}
          onClick={this.handleAddColumnName}
        >
          Добавить колонку
        </button>
        <div
          className="delete-icon remove-column"
          onClick={handleRemoveColumn}
        />
      </div>
    );
  }
}

ColumnHeader.propTypes = {
  handleAddColumn: PropTypes.func.isRequired,
  handleAddColumnClick: PropTypes.func.isRequired,
  handleRemoveColumn: PropTypes.func.isRequired,
};
