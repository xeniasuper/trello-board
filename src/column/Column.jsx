import React from "react";
import {Cards} from "../cards/Cards"
import {ColumnHeader} from "../column_header/ColumnHeader";
import "./column.css";

class Column extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isNameAdded : false,
            cards : null,
        }
    }

    handleAddColumnClick = () => {
        this.setState({
            isNameAdded : true,
            cards : <Cards id={this.props.id}/>
        })
    };

    render() {
        return (
            <div className="column">
                <ColumnHeader handleAddColumnClick={this.handleAddColumnClick}
                              isAdded={this.state.isNameAdded}
                              handleRemoveColumn={() => this.props.removeColumn(this.props.id)}
                />
                {this.state.cards}
            </div>
        )
    }
}

export {Column}
