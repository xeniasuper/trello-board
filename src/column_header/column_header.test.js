import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ColumnHeader from './ColumnHeader';
import Column from '../column/Column';

Enzyme.configure({ adapter: new Adapter() });

it('render correctly board component', () => {
  // const mockHandleAddColumn = jest.fn();
  // const mockHandleAddColumnClick = jest.fn();
  // const mockHandleRemoveColumn = jest.fn();
  // const component = renderer.create(<ColumnHeader
  //                                                 handleAddColumn={mockHandleAddColumn}
  //                                                 handleAddColumnClick={mockHandleAddColumnClick}
  //                                                 handleRemoveColumn={mockHandleRemoveColumn}
  //                                   />);
  // const tree = component.toJSON();
  //
  // expect(tree).toMatchSnapshot();
});

// it('must set state.isNameAdded to true, ' +
//   'column.state.isNameAdded = true, ' +
//   'column.state.cards = Cards id={id}', () => {
//
//   const mockHandleAddColumnClick = jest.fn();
//   const mockHandleRemoveColumn = jest.fn();
//   const columnHeader = shallow(<ColumnHeader
//                                     handleAddColumnClick={mockHandleAddColumnClick}
//                                     handleRemoveColumn={mockHandleRemoveColumn}
//                                 />)
//   }
// })

// it('must call the mock method with p click', () => {
//   const mockCallBack = jest.fn();
//
//   const wrapper = Enzyme.mount(<AddCard createCard={mockCallBack} />);
//   const btn = wrapper.find('#add-one-more-card');
//   btn.simulate('click');
//
//   expect(mockCallBack).toHaveBeenCalled();
// });
