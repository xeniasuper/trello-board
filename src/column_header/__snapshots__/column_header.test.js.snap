// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly board component 1`] = `
<div
  className="column-header"
>
  <div
    className="card column-name"
    contentEditable="true"
    onInput={[Function]}
    placeholder="Введите название колонки"
    spellCheck="false"
  />
  <button
    className="add-column "
    onClick={[Function]}
    type="button"
  >
    Добавить колонку
  </button>
  <div
    className="delete-icon remove-column"
    onClick={[MockFunction]}
  />
</div>
`;
